core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -687,8 +687,22 @@ private[spark] object Utils
         in.close()
       }
     } else {
-      fs.listStatus(path).foreach { fileStatus =>
-        fetchHcfsFile(fileStatus.getPath(), dest, fs, conf, hadoopConf, fileOverwrite)
+      lazy val fsHasPathCapability = {
+        fs.hasPathCapability(path, SparkHadoopUtil.DIRECTORY_LISTING_INCONSISTENT)
+      }
+      val listStatuses = Try {
+        fs.listStatus(path)
+      }
+      listStatuses match {
+        case Failure(e) =>
+          if (e.isInstanceOf[FileNotFoundException] && fsHasPathCapability) {
+            logInfo("Ignoring missing directory %s".format(path))
```

Comment:
For the newly added logs, should we directly use structured logs?

cc @gengliangwang @panbingkun

Reply:
Yes, the (`error`|`warn`|`info`) level logs need to be structured.
please see also:
https://github.com/apache/spark/blob/2516fd8439df42b1c161fbd346a0c346cc075f0f/common/utils/src/test/scala/org/apache/spark/util/StructuredLoggingSuite.scala#L56

Reply:
I'ill fix this